### PR TITLE
feat: Add document format specifier for pdf

### PR DESCRIPTION
**Pull request by koxudaxi**
Related Issues: https://github.com/Mirascope/mirascope/issues/657

### CODE REVIEW

**Comment by koxudaxi** [comment]
@willbakst 
While the issue title specifies implementing `:pdf`, I chose to implement `:document` to maintain consistency with other media types. If you think `:pdf` would be more appropriate, please let me know in the comments and I'll modify it accordingly. (In that case, we might want to rename `DocumentPart` to `PDFPart` as well)

File: docs/learn/prompts.md [hunk 1]
```diff
@@ -198,6 +199,28 @@ Legend: ✓ (Supported), - (Not Supported)
 
     The `:audio` and `:audios` tags support the `bytes | str` and `list[bytes] | list[str]` types, respectively. When passing in a `str`, the string template assumes it indicates a url or local filepath and will attempt to load the bytes from the source.
 
+### Document Inputs
```
Comment (willbakst) [inline docs/learn/prompts.md:202]: note to self: mention this currently only supports PDF format

File: mirascope/core/base/_utils/_get_document_type.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+"""Utility for determining the type of an document from its bytes."""
+
+
+def get_document_type(document_data: bytes) -> str:
+    if document_data.startswith(b"%PDF"):
+        return "pdf"
+    raise ValueError("Unsupported image type")
```
Comment (willbakst) [inline mirascope/core/base/_utils/_get_document_type.py:7]: note to self: update error message to say "document" not "image"

File: mirascope/core/base/_utils/_parse_content_template.py [hunk 3]
```diff
@@ -139,6 +156,18 @@ def _construct_parts(
...
             )
         ]
+    elif part["type"] == "document":
+        source = attrs[part["template"]]
+        return [_construct_document_part(source)] if source else []
+    elif part["type"] == "documents":
+        sources = attrs[part["template"]]
+        if not isinstance(sources, list):
+            raise ValueError(
+                f"When using 'pdfs' template, '{part['template']}' must be a list."
```
Comment (willbakst) [inline mirascope/core/base/_utils/_parse_content_template.py:166]: note to self: update to "documents" not "pdfs"

File: mirascope/core/base/prompt.py [hunk 4]
```diff
@@ -78,6 +78,7 @@ def __str__(self) -> str:
             .replace(":image}", "}")
             .replace(":audios", "}")
             .replace(":audio}", "}")
+            .replace(":pdf}", "}")
```
Comment (willbakst) [inline mirascope/core/base/prompt.py:81]: note to self: replace with "document" and "documents"

**Pull request merged**